src/NWCClient.test.ts

```diff
@@ -56,3 +56,85 @@ describe("NWCClient", () => {
     expect(nwcClient.options.lud16).toBe("hello@getalby.com");
   });
 });
+
+describe("selectHighestCompatibleVersion", () => {
+  let nwcClient: NWCClient;
+  let selectVersion: (walletVersions: string[]) => string | null;
+  const ORIGINAL_SUPPORTED_VERSIONS = NWCClient.SUPPORTED_VERSIONS;
+
+  beforeEach(() => {
+    nwcClient = new NWCClient();
+    // Access the private method using type assertion
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    selectVersion = (nwcClient as any).selectHighestCompatibleVersion;
```

Comment:
Does `selectHighestCompatibleVersion` have to be an instance method? if we moved it out then it can be more easily tested. As far as I can see it can be easily moved out